### PR TITLE
Create GitHub pull requests statistics.

### PR DESCRIPTION
**Pull request by jeschkies**
Summary:
We want to move faster and reduce the number of our pull requests. Our
first step is to generate data and get an overview. The Python script
`ci/github_pulls_stats.py` gets the list of pull requests and calculates
the youngest, oldest, p50 and p90 open pull request.

### CODE REVIEW

File: ci/github_pulls_stats.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env python
```
[the remark this answers is not in the document]
Comment (jeschkies) [reply ci/github_pulls_stats.py]: I'll move it to tools.

[the remark this answers is not in the document]
Comment (zen-dog) [reply ci/github_pulls_stats.py]: +1 on `mv /tools/`

File: ci/github_pulls_stats.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    print("In {:d} pull requests".format(len(ages)))
+    print("Min:", min(ages))
+    print("P50:", ages[p50_index])
+    print("P90:", ages[p90_index])
+    print("Max:", max(ages))
+
+
+if __name__ == "__main__":
+    main()
```
[the remark this answers is not in the document]
Comment (jeschkies) [reply ci/github_pulls_stats.py]: Sofar this is an MVP for the weekly meetings.

File: ci/github_pulls_stats.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+    ages = sorted(ages)
+    p50_index = int(len(ages) * 0.5)
+    p90_index = int(len(ages) * 0.9)
+
+    print("In {:d} pull requests".format(len(ages)))
+    print("Min:", min(ages))
+    print("P50:", ages[p50_index])
+    print("P90:", ages[p90_index])
+    print("Max:", max(ages))
```
[the remark this answers is not in the document]
Comment (jeschkies) [reply ci/github_pulls_stats.py]: This is just a start and not the full story. I'm not sure I know what the open pull request stats mean for one year. We don't have any PRs open for longer than 200 days.

**Comment by jeschkies** [comment]
@zen-dog, what are the action items on my side to merge this pull requests? Your comments hint on the mid term goal but not on the short term task of retrieving the data.

**Comment by zen-dog** [comment]
> what are the action items on my side to merge this pull requests?

agreed.

File: tools/github_pulls_stats.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    now = datetime.now()
+    ages = [now - created_at(pull_request) for pull_request in content]
+
+    ages = sorted(ages)
+    p50_index = int(len(ages) * 0.5)
+    p90_index = int(len(ages) * 0.9)
+
+    print("In {:d} pull requests".format(len(ages)))
+    print("Min:", min(ages))
+    print("P50:", ages[p50_index])
```
Comment (ichernetsky) [inline tools/github_pulls_stats.py]: It will crash if there are no open PRs.

File: tools/github_pulls_stats.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    content = requests.get(uri, params={'state':'open'}).json()
+    now = datetime.now()
+    ages = [now - created_at(pull_request) for pull_request in content]
+
+    ages = sorted(ages)
+    p50_index = int(len(ages) * 0.5)
+    p90_index = int(len(ages) * 0.9)
+
+    print("In {:d} pull requests".format(len(ages)))
+    print("Min:", min(ages))
```
Comment (ichernetsky) [inline tools/github_pulls_stats.py]: Can be just `ages[0]`

File: tools/github_pulls_stats.py [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    print("P50:", ages[p50_index])
+    print("P90:", ages[p90_index])
+    print("Max:", max(ages))
```
Comment (ichernetsky) [inline tools/github_pulls_stats.py]: Can be just `ages[-1]`

Comment (jeschkies) [reply tools/github_pulls_stats.py]: Nice one!

**Comment by zen-dog** [comment]
> Looks like this just gets stats for the current open pull requests?

It's by design - otherwise, we get all possible "old stats" which will skew the stats.

**Pull request merged**